### PR TITLE
Actually document stuff

### LINKED ISSUE #17: Fill out the README
Put more information in the README. @jasonrudolph suggested a section on the tradeoffs and priorities of this package.

### PR DESCRIPTION
**Pull request by smashwilson**
Document the tradeoffs in this library's design and the full public API.

Fixes #17.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,147 @@
...
+`pollingInterval` adjusts the time in milliseconds that the polling thread spends sleeping between polling cycles. Decreasing the interval will improve the timeliness of polled events, but will consume more processor cycles and I/O bandwidth. The interval defaults to `100`.
+
+### watchPath()
+
+Invoke a callback with each batch of filesystem events that occur beneath a specified directory.
+
+```js
+const {watchPath} = require('@atom/watcher')
+const watcher = await watchPath('/var/log', {recursive: true}, (events) => {
+  //
```
Comment (Arcanemagus) [inline README.md]: Maybe put a `...` here?

File: docs/macos.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# MacOS
+
+On MacOS, @atom/watcher uses the [FSEvents API](https://developer.apple.com/library/content/documentation/Darwin/Conceptual/FSEvents_ProgGuide/UsingtheFSEventsFramework/UsingtheFSEventsFramework.html) from Core Foundation. Each registered watch root creates a new `FSEventStream` which is scheduled on the [`CFRunLoop`](https://developer.apple.com/documentation/corefoundation/cfrunloop-rht) associated with the single worker thread. Out-of-band commands are delivered to the worker thread by signalling a [`CFRunLoopSource`](https://developer.apple.com/documentation/corefoundation/1542679-cfrunloopsourcecreate?language=objc).
+
+## fsevents oddities
```
Comment (Arcanemagus) [inline docs/macos.md]: `fsevents` -> `FSEvents`

File: README.md [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 ## Installation
 
-@atom/watcher is developed against node.js 7.4.0 with the `--harmony` flag enabled, but it should work with any node version that supports async/await. Your system must be able to [build native node.js modules](https://github.com/nodejs/node-gyp#installation). @atom/watcher supports [MacOS](./docs/macos.md) _(>= MacOS 10.7)_, [Windows](./docs/windows.md) _(>= Windows XP, >= Windows Server 2003)_, and [Linux](./docs/linux.md) _(kernel >= 2.6.27, glibc >= 2.9 :point_right: Ubuntu >= 10.04, Debian >= 6, RHEL >= 6)_.
+@atom/watcher is developed against Node.js 7.4.0 with the `--harmony` flag enabled, but it should work with any Node.js version that supports `async`/`await`. Your system must be able to [build native node.js modules](https://github.com/nodejs/node-gyp#installation). @atom/watcher supports [MacOS](./docs/macos.md) _(>= MacOS 10.7)_, [Windows](./docs/windows.md) _(>= Windows XP, >= Windows Server 2003)_, and [Linux](./docs/linux.md) _(kernel >= 2.6.27, glibc >= 2.9 :point_right: Ubuntu >= 10.04, Debian >= 6, RHEL >= 6)_.
```
Comment (Arcanemagus) [inline README.md]: `build native node.js modules` -> `build native Node.js modules`

**Comment by smashwilson** [comment]
@Arcanemagus Thanks for the proofreading :smile: 🙇

**Pull request merged**